### PR TITLE
metrics: report: Add net-cpu and in-container-mem tests

### PR DESCRIPTION
**Pull request by grahamwhaley**
Add the network-cpu and in-container-memory consumption tests to the reportgen.
This will allow us to try and identify any:
 -  network CPU work related overheads
 - in-container memory overhead changes.

### CODE REVIEW

**Comment by grahamwhaley** [comment]
@GabyCT - you wrote the in-container memory test, so please have a peek at that :-)
@amshinde - we don't have a huge amount of history (experience) in tracking any changes in the CPU usage/utilisation of Kata running, so you might want to stare at/contemplate/test out the network cpu test.

For reference, here are examples of how the new pages look:

# Memory inside a container:

mem-in

# CPU usage during a fixed bandwidth container<->container test

net-cpu

> And /me spots a typo - there is an 'M' on the y axis in the network picture where that should be a 'G' - let me go find/fix that and push it... ;-)

**Comment by jodh-intel** [comment]
/me catches @grahamwhaley updating the PR he's reviewing... :)

File: metrics/report/report_dockerfile/mem-in-cont.R [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+		datasetname=basename(currentdir)
+
+		# Import the data
+		fdata=fromJSON(fname)
+		fdata=fdata[[testname]]
+		# Copy the average result into a shorter, more accesible name
+		fdata$requested=fdata$Results$memrequest$Result
+		fdata$total=fdata$Results$memtotal$Result
+		fdata$free=fdata$Results$memfree$Result
+		fdata$avail=fdata$Results$memavailable$Result
```
Comment (jodh-intel) [inline metrics/report/report_dockerfile/mem-in-cont.R]: `this$still$makes$my$eyes$bleed$every$time$I$see$it $)`

**Comment by jodh-intel** [comment]
> And /me spots a typo - there is an 'M' on the y axis in the network picture where that should be a 'G' - let me go find/fix that and push it... ;-)

So you found a spot... and it was an M but should be a G? :smile:

File: metrics/report/report_dockerfile/mem-in-cont.R (hunk 1 above)
Comment (grahamwhaley) [reply metrics/report/report_dockerfile/mem-in-cont.R]: :-)
 I could whitespace it more at least, like:
```R
fdata$avail = fdata$Results$memavailable$Result
```

But, the `$`s are R's way of accessing the dataframe members I'm afraid, and the long strong on the RHS is what comes in from the JSON import, so I don't have a huge amount of control over that.

Yeah, they reckon the 80% (of maybe the 80/20 rule) of data analysis is crunching the data into the right format to then be able to process/show it...

**Pull request merged**